sdp/processors/datasets/fleurs/create_initial_manifest.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fnmatch
+import glob
+import json
+import os
+import shutil
+import typing
+from urllib.parse import parse_qs, urlparse
+
+from sdp.processors.base_processor import BaseProcessor, DataEntry
+from sdp.utils.common import download_file, extract_archive
+
+
+def get_fleurs_url_list(lang: str, split: list[str]) -> list[str]:
+    urls = []
+    # examples
+    # "https://huggingface.co/datasets/google/fleurs/resolve/main/data/hy_am/audio/dev.tar.gz",
+    # "https://huggingface.co/datasets/google/fleurs/resolve/main/data/hy_am/dev.tsv"
+
+    base_url = "https://huggingface.co/datasets/google/fleurs/resolve/main/data"
+
+    base_lang_url = os.path.join(base_url, lang)
+    tsv_url = f"{base_lang_url}/{split}.tsv"
+    urls.append(tsv_url)
+
+    tar_gz_url = f"{base_lang_url}/audio/{split}.tar.gz"
+    urls.append(tar_gz_url)
+
+    return urls
+
+
+class CreateInitialManifestFleurs(BaseProcessor):
+    """
+    Processor to create initial manifest for the FLEURS dataset.
+
+    Dataset link: https://huggingface.co/datasets/google/fleurs
+
+    Will download all audio files from the FLEURS dataset, and create a manifest file with
+    the "audio_filepath" and "text" fields.
+
+    Attributes:
+        lang (str): Language to be processed, identified by a combination of ISO 639-1 and ISO 3166-1 alpha-2 codes.
+                    Examples include:
+                    - ``"hy_am"`` for Armenian
+                    - ``"ko_kr"`` for Korean
+        split (str): Which dataset splits to process. Options are:
+                    - ``"test"``
+                    - ``"train"``
+                    - ``"dev"``
+        audio_dir (str): Path to the directory where audio files will be downloaded and extracted.
+    Returns:
+        This processor generates an initial manifest file with the following fields:
+        {
+            "audio_filepath": <path to the audio file>,
+            "text": <transcription>,
+        }
```

Comment:
* Please fix the formatting here - if you build the docs you will see the formatting doesn't look right
* Please also replace "audio_dir" with "raw_data_dir"

Reply:
done